pkg/repo/fs.go

```diff
@@ -210,3 +237,22 @@ func (fsr *FsRepo) WriteRunInfo(ctx context.Context, summaryShellVariablesString
 		}
 	*/
 }
+
+// modifies the config to include keys for accessing repo paths
+func (fsr *FsRepo) EnsureRepoPathsConfigured(c config.ReadWriter) {
+	c.SetIfAbsent(types.AuthTokensPath, fsr.join(config.TokensPath))
+	c.SetIfAbsent(types.UserKeyPath, fsr.join(config.UserPrivateKeyFileName))
+	c.SetIfAbsent(types.NodeExecutorPluginPath, fsr.join(config.PluginsPath))
+
+	// NB(forrest): pay attention to the subtle name difference here
+	c.SetIfAbsent(types.NodeComputeDataPath, fsr.join(config.ComputeStoragesPath))
+	c.SetIfAbsent(types.NodeComputeStoragePath, fsr.join(config.ComputeStorePath))
+
+	c.SetIfAbsent(types.UpdateCheckStatePath, fsr.join(config.UpdateCheckStatePath))
+	c.SetIfAbsent(types.NodeClientAPITLSAutoCertCachePath, fsr.join(config.AutoCertCachePath))
+	c.SetIfAbsent(types.UserLibp2pKeyPath, fsr.join(config.Libp2pPrivateKeyFileName))
+	c.SetIfAbsent(types.NodeNetworkStoreDir, fsr.join(config.OrchestratorStorePath, config.NetworkTransportStore))
+
+	c.SetIfAbsent(types.NodeRequesterJobStorePath, fsr.join(config.OrchestratorStorePath, "jobs.db"))
+	c.SetIfAbsent(types.NodeComputeExecutionStorePath, fsr.join(config.ComputeStorePath, "executions.db"))
+}
```

Comment:
Yeah, this is a bit of :chicken: :egg: problem.
> Config should use the repo as base for the different paths it holds, such as execution and job store paths

This sounds interesting - would it imply the paths in the config are relative paths the repo uses rather than complete paths?

Reply:
default paths are relative to repo path, which is also the case today. Based on the proposal here https://www.notion.so/expanso/Rethinking-Configuration-435fbe87419148b4bbc5119d413786eb, we will only have a repo path and will no longer expose configurations for each individual path